### DISCUSSION TITLE
運用ルール

### DISCUSSION THREAD

**Discussion by light-planck** [light-planck/nemmy#27] [Ideas]
運用ルールについて

**Comment by light-planck** [comment]


**Reply by light-planck** [reply]
別にここじゃなくても良さそうでしたw

**Comment by ghost** [comment]
ブランチ名
対象となる領域(バックエンドかフロントエンドか、それとも両方か)を含める？

**Reply by ghost** [reply]
領域名まで書かなくても衝突はしないかな
(bunは2種類あるけど)

わかりやすさは微妙なところ、長くなるからなー

**Reply by ghost** [reply]
提案しておいてあれですが、自分はいらなさそうだなと思ってます あってもいいですが

**Reply by light-planck** [reply]
ですね、今のところ含めなくて良さそうです！
ありがとうございます！

**Comment by ghost** [comment]
おすすめ拡張機能を`.vscode/extensions.json`に書くのはどうでしょうか！

**Reply by ghost** [reply]
(新メンバーが来なければ不要説はある)

**Reply by ghost** [reply]
参考
https://future-architect.github.io/articles/20200828/

**Reply by light-planck** [reply]
ありがとうございます！
良いと思います！

**Reply by light-planck** [reply]
2人が必要だと思う拡張機能をprで出し合いましょう

**Comment by ghost** [comment]
Biome 使う？

**Reply by light-planck** [reply]
使いテェ😄

**Reply by ghost** [reply]
https://biomejs.dev/formatter/#options
デフォルトでtabインデントなのが神だと思いました👀

**Reply by ghost** [reply]
~~jsのフォーマットをBiomeでやるとして、JSONのフォーマットは引き続きPrettierを使う形になるんですかね~~
JSONのふぉーまっともできるっぽいですね！
https://biomejs.dev/ja/internals/language-support/

**Reply by light-planck** [reply]
私はスペース派閥です😎

**Reply by ghost** [reply]
面白そうなので、採用するのありだと思います！

**Reply by light-planck** [reply]
## メリット

- prettier, eslintの設定ファイルが1つになる
- Rustなので高速
- デフォルトの状態だとeslintよりルールが厳しめ（人によってはデメリットかも）
- prettierでは検知できないコードもフォーマットできる

## デメリット

- ライブラリ専用のeslintが配布されていることがよくある(Next.js, drizzle...)
- 世間的にはprettier + eslintが圧倒的に利用されている
- eslintに対して100%の互換性はない

**Reply by ghost** [reply]
#60

**Comment by light-planck** [comment]
- [ ] コミットメッセージ
- [ ] ブランチ名
- [ ] PR

**Reply by ghost** [reply]
今のところは、プルリクのタイトルに[Conventional Commits](https://www.conventionalcommits.org/ja/v1.0.0/)をゆる〜く適用してる感じですよね

よりきっちりと適用すべきか、プルリクだけでなくコミットメッセージも同じ記法にするか、などを決めていく感じですかね

**Reply by ghost** [reply]
プルリク単位で作業をするなら、各コミットには型(fixやfeatなど)を書かなくてもいいかもなと思いました
プルリクに含まれる各コミットは、プルリクと同じ型の作業になることが多そう(主観)なため

**Reply by light-planck** [reply]
[commitlint](https://github.com/conventional-changelog/commitlint)
のtypeを指定している感じですね！

プルリク単位で作業をしても、コミット全てがそのprと同じtypeになるとは限らない気がします・・・！
例えば、認証機能を追加するprで作業しているときに

```
chore: authライブラリの追加
feat: 認証機能の追加
fix: バグの修正
```

みたいなtypeが想定されます。

typeの指定は任意にするつもりですが、さまざまなOSSでこれと同様のコミット規定がされていることがあるので、typeをつけて損はないと思います！

**Reply by ghost** [reply]
確かにそのとおりですね〜 コミットメッセージにもtypeはつけた方がいいですね！
commitlint 強そう👀
